Code/Framework/AzToolsFramework/AzToolsFramework/Picking/Manipulators/ManipulatorBounds.cpp

```diff
@@ -16,6 +16,83 @@ namespace AzToolsFramework
 {
     namespace Picking
     {
+        // this intersection algorithm is adapted from 'Capped Cone' by Inigo Quilez
+        // ref: https://www.iquilezles.org/www/articles/intersectors/intersectors.htm and https://www.shadertoy.com/view/llcfRf
+        // all algorithms/code snippets are kindly made available under the MIT License - https://www.iquilezles.org/www/index.htm
+        bool IntersectRayCone(
+            const AZ::Vector3& rayOrigin,
+            const AZ::Vector3& rayDirection,
+            const AZ::Vector3& coneApex,
+            const AZ::Vector3& coneAxis,
+            float coneHeight,
+            float coneBaseRadius,
+            float& t)
+        {
+            const AZ::Vector3& pa = coneApex;
+            const AZ::Vector3& pb = coneApex + coneHeight * coneAxis;
+            const AZ::Vector3& ro = rayOrigin;
+            const AZ::Vector3& rd = rayDirection;
+            float rb = coneBaseRadius;
+
+            AZ::Vector3 ba = pb - pa;
+            AZ::Vector3 oa = ro - pa;
+            AZ::Vector3 ob = ro - pb;
+
+            float m0 = ba.Dot(ba);
+            float m1 = oa.Dot(ba);
+            float m2 = ob.Dot(ba);
+            float m3 = rd.Dot(ba);
+
+            auto dot2 = [](const AZ::Vector3& v)
+            {
+                return v.Dot(v);
+            };
+
+            // caps
+            if (m1 < 0.0f)
+            {
+                if (dot2(oa * m3 - rd * m1) < 0.0f)
+                {
+                    t = -m1 / m3;
+                    return true;
+                }
+            }
+            else if (m2 > 0.0f)
+            {
+                if (dot2(ob * m3 - rd * m2) < (rb * rb * m3 * m3))
+                {
+                    t = -m2 / m3;
+                    return true;
+                }
+            }
+
+            // body
+            float m4 = rd.Dot(oa);
+            float m5 = oa.Dot(oa);
+            float rr = -rb;
+            float hy = m0 + rr * rr;
+
+            float k2 = m0 * m0 - m3 * m3 * hy;
+            float k1 = m0 * m0 * m4 - m1 * m3 * hy;
+            float k0 = m0 * m0 * m5 - m1 * m1 * hy;
+
+            float h = k1 * k1 - k2 * k0;
+            if (h < 0.0f)
+            {
+                return false;
+            }
+
+            float tt = (-k1 - AZ::Sqrt(h)) / k2;
```

Comment:
I'm not sure it can be guaranteed that k1 is positive, so I think this could lead to some nasty precision problems if k1 was negative and very close in value to sqrt(h).

Reply:
This may be so but I honestly just want to fix this horrible bug with manipulators and this solves the problem - by moving the code out of AZ::Intersect and using it exclusively with Manipulators I think this is a reasonable option. I just don't have time right now to spend hours and hours debugging the existing function. I wasn't even supposed to be fixing this bug in the first place, I just stumbled across it while working on the line intersection code. I have added a task to come back to this when the other higher priority items are addressed https://github.com/o3de/o3de/issues/3925. If the bug gets fixed in future I'd be happy to switch back to the original implementation.